### PR TITLE
Added support for "additionalProperties" and "allOf" in the UI.

### PR DESCRIPTION
**Pull request by jpeiffer**
Added support within the models for the "additionalProperties" option to render as a proper JS Map and for "allOf" to be able to properly extend models.

### CODE REVIEW

File: lib/swagger-client.js [hunk 1]
```diff
@@ -1478,9 +1481,36 @@ Operation.prototype.encodePathParam = function(pathParam) {
...
+      if (!definition)
+        return;
+      
+      if (definition.allOf) {
+        for (var i=0,len=definition.allOf.length;i<len;i++) {
+          resolve(definition.allOf[i], updatedDefinition, ignoreModels);
+        }
+      } else if (definition.$ref) {
+        var simpleName = simpleRef(definition.$ref);
+        if (typeof ignoreModels[simpleName] !== true) {
```
Comment (mohsen1) [inline lib/swagger-client.js]: what's happening here? `typeof` always return string. This condition is always true.

Comment (jpeiffer) [inline lib/swagger-client.js]: Whoops!  Sorry about that, the "typeof" was supposed to be removed.  I'll get that fixed.

Comment (mohsen1) [inline lib/swagger-client.js]: Please avoid using `for..in` loop and use Underscore methods. We have underscore dependency in Swagger UI

Comment (jpeiffer) [inline lib/swagger-client.js]: Where do you see for..in?  The only instances I see are in the original code, not anything I changed.

**Comment by jpeiffer** [comment]
Checking in to see if there are any concerns with getting this merged.

**Comment by fehguy** [comment]
Hi @jpeiffer we're in the midst of a major refactor of the UI code right now, hence the delay.  It's still open and we'll see how to merge the logic shortly.

I can say though that changes to swagger-client.js should go to the develop_2.0 branch of [swagger-js](https://github.com/swagger-api/swagger-js)

**Comment by jpeiffer** [comment]
I'm not entirely clear on next steps.  Are you planning on incorporating this as part of the refactor, or would you like me to re-apply my changes create a new PR against the develop_2.0 branch of swagger-js?  Either way, it sounds like this PR should be closed.  Agree?

**Comment by fehguy** [comment]
Hi @jpeiffer just track this issue https://github.com/swagger-api/swagger-js/issues/188 as the fix needs to be in swagger-js.

**Pull request closed without merging**